### PR TITLE
Fix bug when path has commas

### PR DESCRIPTION
**Pull request by danielleung27**
#44 Changed delimiter from ',' to '$'

### CODE REVIEW

**Review by churichard — CHANGES_REQUESTED** [review]
This is janky, look at https://stackoverflow.com/a/432353/1744102

File: frontend/src/Search.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ export function updateSchedule(){
         course_entry["semester"] = course.className.split(" ")[1];
         course_entry["dist_area"] = course.getAttribute('dist_area');
         // slice the last element out because it's an empty string: format is "x,y,z,"
```
Comment (churichard) [inline frontend/src/Search.js]: Change comment - it's not relevant anymore, right?

File: frontend/src/Search.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ export function updateSchedule(){
...
-        course_entry["settled"] = course.getAttribute('reqs').split(',').slice(0, -1);
+        course_entry["settled"] = $('.semester').slice(sem_num, sem_num + 1).find('li').slice(course_index, course_index + 1).data('reqs');
```
Comment (churichard) [inline frontend/src/Search.js]: Why are you using slice - can't you just index into the array?

File: frontend/src/Search.js [hunk 3]
```diff
@@ -158,16 +158,15 @@ class Search extends Component {
...
-                if(course['settled'].length > 0) courseReqs = course['settled'].join(',') + ',';
-                return <li key={course["id"]} id={course["id"]} className={"course-display " + course["semester"]} dist_area={course["dist_area"]} reqs={courseReqs}>
+                return <li key={course["id"]} id={course["id"]} className={"course-display " + course["semester"]} dist_area={course["dist_area"]}>
                 <p className="course-name">{course["name"]}</p><i className="fas fa-times-circle delete-course"></i>
                 <p className="course-title">{course["title"]}</p>
                 </li>
               }), document.getElementById('sem' + index))
-              semester.map((course) => {
+              semester.map((course, course_index) => {
+                $('.semester').slice(sem_num, sem_num + 1).find('li').slice(course_index, course_index + 1).data('reqs', course['settled'])
```
Comment (churichard) [inline frontend/src/Search.js]: Why are you using slice - can't you just index into the array?

**Review by churichard — CHANGES_REQUESTED** [review]

File: frontend/src/Search.js (hunk 2 above)
Comment (danielleung27) [reply frontend/src/Search.js]: Indexing into the array was giving me the html element rather than a Jquery selection so I couldn't use .find on it, slicing returns a jquery selection so .find would work. .get() was also giving me an html element. I couldn't find a function to get from index and return a Jquery selection.

Comment (churichard) [reply frontend/src/Search.js]: Try `.eq`: https://api.jquery.com/eq/

File: frontend/src/Search.js [hunk 4]
```diff
@@ -181,7 +179,7 @@ class Search extends Component {
             });
           }
           // get requirements from existing schedule
-          renderRequirements();
+          updateSchedule();
```
Comment (churichard) [inline frontend/src/Search.js:182]: Why did you change this?

**Review by churichard — CHANGES_REQUESTED** [review]

File: frontend/src/Search.js (hunk 4 above)
Comment (danielleung27) [reply frontend/src/Search.js:182]: This fixes the null schedule problem. On startup it should run updateSchedule function so that even if the schedule is empty, it sends back an empty list and sets the user_schedule to equal that empty list. Then the requirements can display. If the schedule is null, the requirements function can't run properly.

**Review by churichard — APPROVED** [review]
